model/src/main/kotlin/licenses/LicenseInfoResolver.kt

```diff
@@ -263,19 +267,23 @@ class LicenseInfoResolver(
                 directories = listOf(directory)
             ).getValue(directory)
 
-            licenseFiles += rootLicenseFiles.map { relativePath ->
-                ResolvedLicenseFile(
+            rootLicenseFiles.forEach { relativePath ->
+                val file = archiveDir.resolve(relativePath)
+                require(file.isValidTextFile()) { "File $relativePath is not a valid text file." }
+                licenseFiles += ResolvedLicenseFile(
                     provenance = provenance,
                     licenseInfo.filter(provenance, relativePath),
                     relativePath,
-                    archiveDir.resolve(relativePath)
+                    file
                 )
             }
         }
 
         return ResolvedLicenseFileInfo(id, licenseFiles)
     }
 
+    private fun File.isValidTextFile() = tika.detect(this).contains("text/", ignoreCase = true)
```

Comment:
This function should be made internal so that dedicated tests just for this function can be added. I would like to have tests for several files based on real world examples to prove that the function generally works as expected.